plugin-server/src/worker/rusty-hook.ts

```diff
@@ -122,4 +127,49 @@ export class RustyHook {
 
         return true
     }
+
+    public async enqueueForHog(payload: HogFunctionInvocationResult): Promise<boolean> {
+        // This is a temporary copy of `enqueueIfEnabledForTeam` above for Hog fetches because the
+        // API differs. It will likely be replaced with a Kafka topic soon.
+
+        const body = JSON.stringify(payload)
+
+        // We attempt to enqueue into the rusty-hook service until we succeed. This is deliberatly
+        // designed to block up the consumer if rusty-hook is down or if we deploy code that
+        // sends malformed requests. The entire purpose of rusty-hook is to reliably deliver webhooks,
+        // so we'd rather leave items in the Kafka topic until we manage to get them into rusty-hook.
+        let attempt = 0
+        while (true) {
+            try {
+                attempt += 1
+                const response = await fetch(this.serverConfig.HOG_HOOK_URL, {
+                    method: 'POST',
+                    headers: { 'Content-Type': 'application/json' },
+                    body,
+
+                    // Sure, it's not an external request, but we should have a timeout and this is as
+                    // good as any.
+                    timeout: this.serverConfig.EXTERNAL_REQUEST_TIMEOUT_MS,
+                })
+
+                if (response.ok) {
+                    // Success, exit the loop.
+                    break
+                }
+
+                // Throw to unify error handling below.
+                throw new Error(
```

Comment:
Running locally - I got this error a lot - `"message": "network timeout at: http://localhost:3300/hoghook",`

I setup a couple of hog functions locally that are designed to just hammer the api in a loop to simulate sending every event somewhere. Especially if I didn't run the plugins server for a while to allow it to build up a big backlog. When it was working through full 500 messsage batches, eventually the webhook service would just seem to stop responding.

Might be worth looking at it together to try and recreate the issue as I don't know if this is just a limit of local testing or something actually going wrong.